File: 151004/Tishalovich/rest_basics/src/main/java/by/bsuir/dc/rest_basics/entities/Author.java
package by.bsuir.dc.rest_basics.entities;

import jakarta.persistence.*;
import lombok.*;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tblAuthor")
public class Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String login;

    private String password;

    private String firstName;

    private String lastName;

    @OneToMany(mappedBy = "author")
    private List<Story> stories;
}

File: 151004/Sakovsky/publisher/src/entities/Author.ts
import { Column, Entity, JoinColumn, JoinTable, OneToMany, PrimaryGeneratedColumn } from "typeorm";
import { Tweet } from "./Tweet";

@Entity({name: 'tbl_authors'})
export class Author {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({unique: true})
    login: string;

    @Column()
    password: string;

    @Column()
    firstname: string;

    @Column()
    lastname: string;

    @OneToMany(()=>Tweet, tweet => tweet.author)
    tweets: Tweet[];
}